app/services/preserved_object_handler.rb

```diff
@@ -272,14 +267,19 @@ def update_pc_invalid_moab
     handler_results.remove_db_updated_results unless transaction_ok
   end
 
-  def check_status_if_not_ok(pres_copy, found_expected_version)
-    if pres_copy.status != PreservedCopy::OK_STATUS
+  # TODO: what if this checked status in the case of version mismatcch also?  would that simplify
+  # check_existence and/or confirm_version in their handling after calling this?  could they just
+  # deal with the result code wranging after calling this if it also checked status on vers mismatch?
```

Comment:
The question you raise here is worthy - I wonder if it should be a refactor ticket rather than a comment here?  Or in addition?  Or maybe worth no more than 1 hour's experimentation?